src/migrate/Migrator.js

```diff
@@ -419,15 +438,22 @@ export default class Migrator {
   }
 
   setConfig(config) {
-    return assign({}, CONFIG_DEFAULT, this.config || {}, config);
+    const newConfig = assign({}, CONFIG_DEFAULT, this.config || {}, config);
+    if ((config && !config.migrationSource) || !newConfig.migrationSource) {
```

Comment:
Why checking just newConfig is not enough?

Reply:
Because of the way tests use this function, I only want to init the migrationSource if it is not on the current config (if there is one) and it's also not specified in the new config.

Just checking new config meant that different tests calling setConfig would not apply.

Reply:
Not sure if I understand. After you call `assign({}, CONFIG_DEFAULT, this.config || {}, config);`, doesn't newConfig already include everything that config has?

Reply:
Let's say the first test does not specify a migrationSource, but sets the directory. The migrationSource will capture the directory, so when setConfig is called with a different directory, but no migrationSource the current migrationSource (with the old directory) is on this.config.

Basically, the tests can't change the migration directory.

Reply:
Looks like we are talking about different things :)
Under which circumstances would "!config.migrationSource" check pass but " !newConfig.migrationSource" would not? (double check seems redundant to me is what I'm saying as one config is derived from the other)

Reply:
If you were just to check newConfig.migrationSource, this would happen

```
const migrator = new Migrator({ directory: '/foo' })
migrator.config.migrationSource.directory === '/foo'

migrator.setConfig({ directory: '/bar' })
// !config.migrationSource === true
migrator.config.migrationSource.directory === '/foo'
```

migrationSource can be set by the user, but also falls back. If the user set a migration source the first time, we don't want to remove it. Will see if I can clean up somehow. But without the extra logic, heaps of tests fail because they can't change the directory.

Reply:
Hrrmm, looking at this fresh today. I am not sure exactly.

I *think* I was thinking something along the lines of, if you specify a migration source initially, which is not the default, I don't want it overridden if you call setConfig again.

Either way seems brittle, going to try and figure a better way

Reply:
Have re-written logic so it's clearer